pkg/solver/store/memory/store.go

```
@@ -197,6 +197,17 @@ func (s *SolverStoreMemory) GetResourceOffer(id string) (*data.ResourceOfferCont
 	return resourceOffer, nil
 }
 
+func (s *SolverStoreMemory) GetResourceOfferByAddress(address string) (*data.ResourceOfferContainer, error) {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+	for _, resourceOffer := range s.resourceOfferMap {
+		if resourceOffer.ResourceProvider == address {
+			return resourceOffer, nil
+		}
+	}
+	return nil, nil
```

Could we return an error here? We observed some issues handling errors at the call site:

https://github.com/Lilypad-Tech/lilypad/blob/c419ac1159633237275e80ad55a3c8b7613f980b/pkg/solver/matcher.go#L150-L153

Perhaps we could print the error in this block and continue.